### ISSUE TITLE
plot_directive seems to be partly ignoring plot_formats

### ISSUE THREAD

**Issue by ibell** [matplotlib/matplotlib#2800]
I have a project with hundreds (~500) of figures that are generated by MPL.  I have a command in the conf.py to tell MPL to only generate PNG files at 80 dpi

```
plot_formats = [('png',80)]
```

but when I run the build, I get an error like

```
C:\Users\Belli\Documents\Code\CoolProp\Web\Fluids\trans-2-Butene.rst:19: WARNING: image file not readable: Fluids\../_build/plot_directive/Fluids/trans-2-Butene-4.pdf
```

for every single figure in my project. 

Of course the PDF isn't readable since it shouldn't be being checked in the first place!  I believe this is a regression since 1.2 or so (not sure of the exact revision though).

Hope a guru can chime in with wisdom!

**Comment by mgeier** [comment]
I believe this was fixed in #3530. A similar problem still exists for PNG files (or lack thereof), see #13633.

**Comment by QuLogic** [comment]
This should have been fixed by the above-mentioned PRs.
